### DISCUSSION TITLE
Raising the visibility of our roadmap and making it open to participation

### DISCUSSION THREAD

**Discussion by psd** [digital-land/digital-land#350] [General]
We want to make our roadmap items: prototypes, projects, services, missions, activities and definitions of success publicly accessible. This is so we can:

* raise the visibility of our work inside and outside our organisation
* help users and stakeholders understand what we're working on _particularly for users of our data design service who work on policy, with grids and timelines_
* lower the barrier for others to help us by participating in our design decisions _one of our principles is to be inclusive when designing_
* reduce the burden on the team answering the needs of governance within our organisation 
* meet other external governance, such as the [digital service standard](https://www.gov.uk/service-manual/service-standard).

We need to find the users and research what they need to know. 

In the meantime, what exemplary roadmaps should we try and emulate?

**Comment by psd** [comment]
Neil Williams gave a talk at mind the product on [early experiments in roadmapping at GDS](https://www.mindtheproduct.com/2014/07/experiments-roadmapping-gov-uk/) also on the [GDS blog](https://insidegovuk.blog.gov.uk/2014/07/31/experiments-in-roadmapping-at-gov-uk/) [PDF](http://cdn02.mindtheproduct.com/wp-content/uploads/2014/07/Roadmap.10.07.14_redact.pdf)

![PDF](https://66.media.tumblr.com/5267556d20f220e1e1ce905822eb64c8/tumblr_pjlb1g0CyC1qz4bcto1_640.png)

The [GOV.UK 2016-17 roadmap](https://insidegovuk.blog.gov.uk/2016/06/03/the-gov-uk-roadmap-2016-to-2017/) became much more GANTT-like. I'm not a fan of how certain this feels, but we could make a similar visualisation to help show a timeline of each project.

![GANTT](https://insidegovuk.blog.gov.uk/wp-content/uploads/sites/24/2016/06/Roadmap-screenshot.png)

The [2017 vision for GOV.UK](https://insidegovuk.blog.gov.uk/2016/03/09/the-vision-for-gov-uk-is-to-make-government-work-for-users/) looks like something we should emulate.

The [2017-18 GOV.UK roadmap](https://insidegovuk.blog.gov.uk/2017/02/13/the-2017-to-2018-gov-uk-roadmap/) has a stronger vision and narrative, but still uses product plan.

**Comment by psd** [comment]
@jamiearnold has some great posts on the practice behind creating a roadmap:

[Everyone loves a roadmap](http://www.jamiearnold.com/blog/2014/07/07/everyone-loves-a-roadmap) explains why we should work at this, and [seven questions to build a roadmap](http://www.jamiearnold.com/blog/2014/07/22/seven-questions-to-build-a-roadmap) has a great approach for how we can develop our plan:

1. What are we trying to prove or learn?
2. Who are the users?
3. What are we operating?
4. What are we saying?
5. What are our assumptions?
6. What are the dependencies?
7. What capabilities do we need?

![roadmaps](https://static1.squarespace.com/static/5691203fdf40f361d6ec5769/t/588c98bc9f74563ea53ffd5d/1485609156546/?format=1000w)

[![Jamie's agile drawing](https://gds.blog.gov.uk/wp-content/uploads/sites/60/2015/06/jamie-agile-drawing.jpg)](https://gds.blog.gov.uk/2015/06/30/outcomes-not-deliverables/)

**Comment by psd** [comment]
Richard Pope showed a roadmap in his anatomy of a project space:

[![Project Space](https://richardpope.org/images/blog/anatomy-of-a-project-space.jpg)](https://richardpope.org/blog/2014/06/22/anatomy-of-a-project-space/)

**Comment by psd** [comment]
Here's the 2018 Q3 roadmap slide we made for internal governance purposes:

![q4-roadmap](https://user-images.githubusercontent.com/5833/49868058-2b9f8900-fe04-11e8-95fa-cec8212cf681.jpg)

We were particularly weak on how we'll measure outcomes during our alpha.

**Comment by psd** [comment]
@yahoopete talked about his team's experiences running an open backlog when [delivering inside-government](https://gds.blog.gov.uk/2012/04/30/delivering-inside-government/)

**Comment by psd** [comment]
We should turn https://digital-land.github.io/ into a roadmap service, built using data and content in this issues list and repository, a place to find all the things:

[![roadmap](https://user-images.githubusercontent.com/5833/49881127-7b427c80-fe25-11e8-8c65-0a896f6132eb.jpg)](
https://flic.kr/p/2aLygxj)

**Comment by psd** [comment]
More sketching about being clearer on our strategy: 

<a href="https://www.flickr.com/photos/psd/45642277884/in/dateposted-ff/" title="Back of the bus roadmap thinking"><img src="https://farm5.staticflickr.com/4855/45642277884_8d9389ac0d_c.jpg" alt="Back of the bus roadmap thinking"></a>

Being clearer how the strategy is how we get from where we are to where we want to be. Roadmap seems to be used for quarterly planning, which is a small part of the story we should tell for good governance.

**Comment by psd** [comment]
The GOV.UK transformation programme had a dashboard of projects and their status:

![image](https://user-images.githubusercontent.com/5833/55175877-0ac32f80-5178-11e9-9709-7f5eead85d7f.png)
![image](https://user-images.githubusercontent.com/5833/55176042-4b22ad80-5178-11e9-85a2-12b824c86022.png)


https://webarchive.nationalarchives.gov.uk/20150204082752/https://www.gov.uk/transformation

This has been archived in a more vanilla format: https://www.gov.uk/government/publications/gds-transformation-programme-2013-to-2015/digital-transformation-programme

**Comment by psd** [comment]
@Ellayanor pointed at architecture decision records as a way of recording design decisions: https://crown-commercial-service.github.io/ReportMI-service-manual/#/decision-records?id=product-decision-records

**Comment by jakemulley** [comment]
Examples: [Making Monzo][mm], [Office 365 Roadmap][msroadmap]

It's definitely worth having a public, simple roadmap that showcases a high-level overview of what the team is doing similar to the examples above, sticking to three stages, using non-commital language, such as "planned", "in progress", and "done", rather than [Microsoft's][msroadmap] "In development", "Rolling out" and "Launched" to help us when priorities change. The terms used by Microsoft are very commital and are linked to Q* release dates. Some work has already been done on [Digital Land's project pages][dl].

The items in the roadmap stages can then link to a more low-level descriptive piece and can further provide a way for those who are engaged in our work to publicly discuss and comment on.

To help raise visibility of our work, both inside and outside our organisation, we should steer clear of using something like GitHub to track projects, as it is developer-centric and it may be worth using more common software, such as Trello, which may be easier for non-technical people to use.

A key piece around this work is absolute transparency, including [documenting decisions][dd] and talking openly about what work has been prioritised, and why we're focusing on it. This should include things such as user research and their outcomes, in particular helping Local Authorities or other government departments, devolved or otherwise, who may be looking to do similar work to Digital Land. This can also aid with helping users and stakeholders understand what we're working on and why, alongside lowering barriers for others to get involved.

It is also worth including community guidelines around this to help keep discussions topical and suited to needs of those both in and outside the organisation.

[mm]: https://monzo.com/community/making-monzo
[msroadmap]: https://www.microsoft.com/en-gb/microsoft-365/roadmap
[dd]: https://github.com/digital-land/digital-land/issues/36#issuecomment-477712943
[dl]: https://digital-land.github.io/project/

**Comment by psd** [comment]
Martin Weigel's [Strategy needs good words](https://martinweigel.org/2019/03/20/strategy-needs-good-words/) has great advice on writing a strategy:

  1. Know your audience
  2. Make action your goal
  3. Narrative demands jeopardy
  4. Structure creates narrative
  5. There is only momentum or useful description
  6. Narrative travels in one direction only
  7. Make no assumptions
  8. Be the ruthless editor
  9. Try not to create in slideware
 10. 10 Use small words to express big ideas
 11. Don’t hide behind jargon
 12. Make time for deep work
 13. Read good words

**Comment by psd** [comment]
This thread from [Jana Bastow](https://twitter.com/simplybastow) beautifully argues against timeline based roadmaps: https://twitter.com/simplybastow/status/1168531672335343616

I particularly like the prodpad vision template:
<a href="https://www.prodpad.com/blog/product-vision-template/"><img src="https://pbs.twimg.com/media/EDd1Z5pXsAA1R6C.jpg"></a>

raising objectives:
<a href="https://twitter.com/simplybastow/status/1168531737766432768"><img src="https://pbs.twimg.com/media/EDd1aVgWkAEvK0h.jpg"></a>

the time horizons:
<a href="https://twitter.com/simplybastow/status/1168531721786134528"><img src="https://pbs.twimg.com/media/EDd1avQWsAEVrrz.jpg"></a>

and the lean road-mapping canvas:
<a href="https://twitter.com/simplybastow/status/1168531746427678721"><img src="https://pbs.twimg.com/media/EDd1bT5WkAE9Mf3.jpg"></a>

**Comment by psd** [comment]
GOV.UK has published: https://www.gov.uk/roadmap
![screencapture-gov-uk-roadmap-2021-02-02-11_16_40](https://user-images.githubusercontent.com/5833/106593087-44210f00-6548-11eb-9952-2af3f768855e.png)

**Comment by psd** [comment]
The GOV.UK PaaS team used trello for its roadmap: https://github.com/alphagov/paas-roadmap

**Comment by psd** [comment]
Wikidata has a development plan: https://www.wikidata.org/wiki/Wikidata:Development_plan
![wikidata roadmap](https://upload.wikimedia.org/wikipedia/commons/9/92/Wikidata_goals_21.jpg)

**Comment by psd** [comment]
DARPA has [The Heilmeier Catechism](https://www.darpa.mil/work-with-us/heilmeier-catechism) for new projects, good questions for a roadmap to answer:

- What are you trying to do? Articulate your objectives using absolutely no jargon.
- How is it done today, and what are the limits of current practice?
- What is new in your approach and why do you think it will be successful?
- Who cares? If you are successful, what difference will it make?
- What are the risks?
- How much will it cost?
- How long will it take?
- What are the mid-term and final “exams” to check for success?

**Comment by paulmsmith** [comment]
Simple Analytics have a nice open road map:

![image](https://user-images.githubusercontent.com/57055/115544630-be854380-a29a-11eb-853f-89f8e695866f.png)

It is a different model from us but I like how it is linked to tickets in Github.

https://simpleanalytics.com/roadmap

**Comment by paulmsmith** [comment]
Ben Holliday's '[Framing the problem' questions have](https://www.hollidazed.co.uk/2015/07/28/frame-the-problem/) been a go to set of questions for any work I've done.

![image](https://user-images.githubusercontent.com/57055/115545122-54b96980-a29b-11eb-9933-a534eefdec31.png)

**Comment by paulmsmith** [comment]
> The GOV.UK PaaS team used trello for its roadmap: https://github.com/alphagov/paas-roadmap

I like the idea that whatever we use is part of our workflow and adds little-to-no additional burden and has a low barrier to entry.

So a Trello board that our sprint board could reference might work. Trello is a lot less scary and technical looking that Github too and can be styled slightly too.

**Comment by paulmsmith** [comment]
@stevenjmesser shared this [on Twitter](https://twitter.com/stevenjmesser/status/1364507298320834560) re: roadmaps:

![image](https://user-images.githubusercontent.com/57055/115837293-535d7d80-a410-11eb-8e6f-18e9ab92dba4.png)

**Comment by stevenjmesser** [comment]
> @stevenjmesser shared this [on Twitter](https://twitter.com/stevenjmesser/status/1364507298320834560) re: roadmaps:
> 
> ![image](https://user-images.githubusercontent.com/57055/115837293-535d7d80-a410-11eb-8e6f-18e9ab92dba4.png)

Thanks for referencing it, @paulmsmith! I wrote this on [open collaborative roadmaps](https://visitmy.website/2020/07/10/open-collaborative-roadmaps/) in case it's helpful.

**Comment by paulmsmith** [comment]
@stevenjmesser Awesome! Thanks for writing it!

**Comment by psd** [comment]
I really like how [GOV.UK Pay](https://www.payments.service.gov.uk/roadmap/) and [GOV.UK Notify](https://www.notifications.service.gov.uk/features) frame their roadmaps as upcoming features. This clarifies the reason for a public roadmap is to help users and stakeholders plan and make decisions based on the intent of the team.

**Comment by psd** [comment]
[James Higgott on creating the NHS website roadmap](https://jiggott.medium.com/creating-the-nhs-website-roadmap-a-case-study-2ad98d6087d2)

![image](https://user-images.githubusercontent.com/5833/227996359-d58f0c85-5467-4095-840c-c14d2f0785f9.png)

**Comment by stevenjmesser** [comment]
Funny to see some of my bumpf up there 😅

If we use GitHub Projects to manage our work, it has a [roadmap view](https://github.blog/2022-11-15-the-journey-of-your-work-has-never-been-clearer/#roadmaps-to-easily-see-where-youre-going) of in-progress and upcoming work.

**Comment by stevenjmesser** [comment]
I’ve created a [platform roadmap](https://github.com/orgs/digital-land/projects/11) in GitHub Projects and signed up to GitHub’s public beta for [sub-issues and types](https://github.blog/changelog/2024-10-01-evolving-github-issues-public-beta/). This will allow us to create issues for each of our OKRs, and teams can create sub-issues for their work – linking the work done on teams to our OKRs. We can then link the objectives on our [roadmap page](https://www.planning.data.gov.uk/about/roadmap) through to the work inside GitHub Projects – proper working in the open.

**Comment by stevenjmesser** [comment]
I like the idea of using GitHub Discussions more too, @psd, thanks for moving the issue here. For some aspects of our work, we could start a discussion and invite people inside and outside the teams to comment.
